compile error with datacollection.hpp on windows

On Windows, inside Ascent, I see the following compile error:
```
 C:\Users\cyrus\Documents\GitHub\ascent\scripts\build_ascent\install\mfem-4.5\include\mfem\fem\datacollection.hpp(380,11): error C2059: syntax error: 'constant'
```

The related code is an enum def:

https://github.com/mfem/mfem/blob/d5f46a61bc77df5692527bf827a3829f512cd28c/fem/datacollection.hpp#L381


Even though this is an enum which is inside a class and a namespace -- I suspect I have some include that brings in a macro that conflicts with one of these names. I am going to try to track the include issue down - but if you are open to using a name unique to MFEM (`MFEM_NO_ERROR` ?) that should solve the issue.

--- a/fem/datacollection.cpp
+++ b/fem/datacollection.cpp
@@ -111,7 +111,7 @@ DataCollection::DataCollection(const std::string& collection_name, Mesh *mesh_)
    pad_digits_cycle = pad_digits_rank = pad_digits_default;
    format = SERIAL_FORMAT; // use serial mesh format
    compression = 0;
-   error = NO_ERROR_;
+   error = No_Error;
 }
 
 void DataCollection::SetMesh(Mesh *new_mesh)
@@ -494,7 +494,7 @@ void VisItDataCollection::Load(int cycle_)
 {
    DeleteAll();
    time_step = 0.0;
-   error = NO_ERROR_;
+   error = No_Error;
    cycle = cycle_;
    std::string root_name = prefix_path + name + "_" +
                            to_padded_string(cycle, pad_digits_cycle) +

--- a/fem/datacollection.cpp
+++ b/fem/datacollection.cpp
@@ -111,7 +111,7 @@ DataCollection::DataCollection(const std::string& collection_name, Mesh *mesh_)
    pad_digits_cycle = pad_digits_rank = pad_digits_default;
    format = SERIAL_FORMAT; // use serial mesh format
    compression = 0;
-   error = NO_ERROR;
+   error = NO_ERROR_;
 }
 
 void DataCollection::SetMesh(Mesh *new_mesh)
@@ -494,7 +494,7 @@ void VisItDataCollection::Load(int cycle_)
 {
    DeleteAll();
    time_step = 0.0;
-   error = NO_ERROR;
+   error = NO_ERROR_;
    cycle = cycle_;
    std::string root_name = prefix_path + name + "_" +
                            to_padded_string(cycle, pad_digits_cycle) +

--- a/fem/datacollection.hpp
+++ b/fem/datacollection.hpp
@@ -378,12 +378,24 @@ public:
    virtual ~DataCollection();
 
    /// Errors returned by Error()
-   enum { NO_ERROR = 0, READ_ERROR = 1, WRITE_ERROR = 2 };
+   enum
+   {
+      // Workaround for use with headers that define NO_ERROR as a macro,
+      // e.g. winerror.h (which is included by Windows.h):
+#ifndef NO_ERROR
+      NO_ERROR = 0,
+#endif
+      // Use the following identifier if NO_ERROR is defined as a macro,
+      // e.g. winerror.h (which is included by Windows.h):
+      NO_ERROR_   = 0,
+      READ_ERROR  = 1,
+      WRITE_ERROR = 2
+   };
 
    /// Get the current error state
    int Error() const { return error; }
    /// Reset the error state
-   void ResetError(int err_state = NO_ERROR) { error = err_state; }
+   void ResetError(int err_state = NO_ERROR_) { error = err_state; }
 
 #ifdef MFEM_USE_MPI
    friend class ParMesh;

--- a/fem/datacollection.hpp
+++ b/fem/datacollection.hpp
@@ -387,7 +387,7 @@ public:
 #endif
       // Use the following identifier if NO_ERROR is defined as a macro,
       // e.g. winerror.h (which is included by Windows.h):
-      NO_ERROR_   = 0,
+      No_Error    = 0,
       READ_ERROR  = 1,
       WRITE_ERROR = 2
    };
@@ -395,7 +395,7 @@ public:
    /// Get the current error state
    int Error() const { return error; }
    /// Reset the error state
-   void ResetError(int err_state = NO_ERROR_) { error = err_state; }
+   void ResetError(int err_state = No_Error) { error = err_state; }
 
 #ifdef MFEM_USE_MPI
    friend class ParMesh;

--- a/general/mem_manager.cpp
+++ b/general/mem_manager.cpp
@@ -786,7 +786,7 @@ void *MemoryManager::New_(void *h_tmp, size_t bytes, MemoryType h_mt,
    void *h_ptr;
    if (h_tmp == nullptr) { ctrl->Host(h_mt)->Alloc(&h_ptr, bytes); }
    else { h_ptr = h_tmp; }
-   flags = Mem::REGISTERED_ | Mem::OWNS_INTERNAL | Mem::OWNS_HOST |
+   flags = Mem::Registered | Mem::OWNS_INTERNAL | Mem::OWNS_HOST |
            Mem::OWNS_DEVICE | valid_flags;
    // The other New_() method relies on this lazy allocation behavior.
    mm.Insert(h_ptr, bytes, h_mt, d_mt); // lazy dev alloc
@@ -820,7 +820,7 @@ void *MemoryManager::Register_(void *ptr, void *h_tmp, size_t bytes,
       return nullptr;
    }
 
-   flags |= Mem::REGISTERED_ | Mem::OWNS_INTERNAL;
+   flags |= Mem::Registered | Mem::OWNS_INTERNAL;
    void *h_ptr;
 
    if (is_host_mem) // HOST TYPES + MANAGED
@@ -859,7 +859,7 @@ void MemoryManager::Register2_(void *h_ptr, void *d_ptr, size_t bytes,
       return;
    }
 
-   flags |= Mem::REGISTERED_ | Mem::OWNS_INTERNAL;
+   flags |= Mem::Registered | Mem::OWNS_INTERNAL;
 
    MFEM_VERIFY(d_ptr || bytes == 0,
                "cannot register NULL device pointer with bytes = " << bytes);
@@ -911,7 +911,7 @@ void MemoryManager::SetDeviceMemoryType_(void *h_ptr, unsigned flags,
 void MemoryManager::Delete_(void *h_ptr, MemoryType h_mt, unsigned flags)
 {
    const bool alias = flags & Mem::ALIAS;
-   const bool registered = flags & Mem::REGISTERED_;
+   const bool registered = flags & Mem::Registered;
    const bool owns_host = flags & Mem::OWNS_HOST;
    const bool owns_device = flags & Mem::OWNS_DEVICE;
    const bool owns_internal = flags & Mem::OWNS_INTERNAL;
@@ -1018,7 +1018,7 @@ void *MemoryManager::ReadWrite_(void *h_ptr, MemoryType h_mt, MemoryClass mc,
                                 size_t bytes, unsigned &flags)
 {
    if (h_ptr) { CheckHostMemoryType_(h_mt, h_ptr, flags & Mem::ALIAS); }
-   if (bytes > 0) { MFEM_VERIFY(flags & Mem::REGISTERED_,""); }
+   if (bytes > 0) { MFEM_VERIFY(flags & Mem::Registered,""); }
    MFEM_ASSERT(MemoryClassCheck_(mc, h_ptr, h_mt, bytes, flags),"");
    if (IsHostMemory(GetMemoryType(mc)) && mc < MemoryClass::DEVICE)
    {
@@ -1042,7 +1042,7 @@ const void *MemoryManager::Read_(void *h_ptr, MemoryType h_mt, MemoryClass mc,
                                  size_t bytes, unsigned &flags)
 {
    if (h_ptr) { CheckHostMemoryType_(h_mt, h_ptr, flags & Mem::ALIAS); }
-   if (bytes > 0) { MFEM_VERIFY(flags & Mem::REGISTERED_,""); }
+   if (bytes > 0) { MFEM_VERIFY(flags & Mem::Registered,""); }
    MFEM_ASSERT(MemoryClassCheck_(mc, h_ptr, h_mt, bytes, flags),"");
    if (IsHostMemory(GetMemoryType(mc)) && mc < MemoryClass::DEVICE)
    {
@@ -1066,7 +1066,7 @@ void *MemoryManager::Write_(void *h_ptr, MemoryType h_mt, MemoryClass mc,
                             size_t bytes, unsigned &flags)
 {
    if (h_ptr) { CheckHostMemoryType_(h_mt, h_ptr, flags & Mem::ALIAS); }
-   if (bytes > 0) { MFEM_VERIFY(flags & Mem::REGISTERED_,""); }
+   if (bytes > 0) { MFEM_VERIFY(flags & Mem::Registered,""); }
    MFEM_ASSERT(MemoryClassCheck_(mc, h_ptr, h_mt, bytes, flags),"");
    if (IsHostMemory(GetMemoryType(mc)) && mc < MemoryClass::DEVICE)
    {
@@ -1088,8 +1088,8 @@ void MemoryManager::SyncAlias_(const void *base_h_ptr, void *alias_h_ptr,
                                size_t alias_bytes, unsigned base_flags,
                                unsigned &alias_flags)
 {
-   // This is called only when (base_flags & Mem::REGISTERED_) is true.
-   // Note that (alias_flags & REGISTERED_) may not be true.
+   // This is called only when (base_flags & Mem::Registered) is true.
+   // Note that (alias_flags & Registered) may not be true.
    MFEM_ASSERT(alias_flags & Mem::ALIAS, "not an alias");
    if ((base_flags & Mem::VALID_HOST) && !(alias_flags & Mem::VALID_HOST))
    {
@@ -1097,10 +1097,10 @@ void MemoryManager::SyncAlias_(const void *base_h_ptr, void *alias_h_ptr,
    }
    if ((base_flags & Mem::VALID_DEVICE) && !(alias_flags & Mem::VALID_DEVICE))
    {
-      if (!(alias_flags & Mem::REGISTERED_))
+      if (!(alias_flags & Mem::Registered))
       {
          mm.InsertAlias(base_h_ptr, alias_h_ptr, alias_bytes, base_flags & Mem::ALIAS);
-         alias_flags = (alias_flags | Mem::REGISTERED_ | Mem::OWNS_INTERNAL) &
+         alias_flags = (alias_flags | Mem::Registered | Mem::OWNS_INTERNAL) &
                        ~(Mem::OWNS_HOST | Mem::OWNS_DEVICE);
       }
       mm.GetAliasDevicePtr(alias_h_ptr, alias_bytes, true);
@@ -1671,7 +1671,7 @@ void MemoryPrintFlags(unsigned flags)
 {
    typedef Memory<int> Mem;
    mfem::out
-         << "\n   registered    = " << bool(flags & Mem::REGISTERED_)
+         << "\n   registered    = " << bool(flags & Mem::Registered)
          << "\n   owns host     = " << bool(flags & Mem::OWNS_HOST)
          << "\n   owns device   = " << bool(flags & Mem::OWNS_DEVICE)
          << "\n   owns internal = " << bool(flags & Mem::OWNS_INTERNAL)

--- a/general/mem_manager.cpp
+++ b/general/mem_manager.cpp
@@ -786,7 +786,7 @@ void *MemoryManager::New_(void *h_tmp, size_t bytes, MemoryType h_mt,
    void *h_ptr;
    if (h_tmp == nullptr) { ctrl->Host(h_mt)->Alloc(&h_ptr, bytes); }
    else { h_ptr = h_tmp; }
-   flags = Mem::REGISTERED | Mem::OWNS_INTERNAL | Mem::OWNS_HOST |
+   flags = Mem::REGISTERED_ | Mem::OWNS_INTERNAL | Mem::OWNS_HOST |
            Mem::OWNS_DEVICE | valid_flags;
    // The other New_() method relies on this lazy allocation behavior.
    mm.Insert(h_ptr, bytes, h_mt, d_mt); // lazy dev alloc
@@ -820,7 +820,7 @@ void *MemoryManager::Register_(void *ptr, void *h_tmp, size_t bytes,
       return nullptr;
    }
 
-   flags |= Mem::REGISTERED | Mem::OWNS_INTERNAL;
+   flags |= Mem::REGISTERED_ | Mem::OWNS_INTERNAL;
    void *h_ptr;
 
    if (is_host_mem) // HOST TYPES + MANAGED
@@ -859,7 +859,7 @@ void MemoryManager::Register2_(void *h_ptr, void *d_ptr, size_t bytes,
       return;
    }
 
-   flags |= Mem::REGISTERED | Mem::OWNS_INTERNAL;
+   flags |= Mem::REGISTERED_ | Mem::OWNS_INTERNAL;
 
    MFEM_VERIFY(d_ptr || bytes == 0,
                "cannot register NULL device pointer with bytes = " << bytes);
@@ -911,7 +911,7 @@ void MemoryManager::SetDeviceMemoryType_(void *h_ptr, unsigned flags,
 void MemoryManager::Delete_(void *h_ptr, MemoryType h_mt, unsigned flags)
 {
    const bool alias = flags & Mem::ALIAS;
-   const bool registered = flags & Mem::REGISTERED;
+   const bool registered = flags & Mem::REGISTERED_;
    const bool owns_host = flags & Mem::OWNS_HOST;
    const bool owns_device = flags & Mem::OWNS_DEVICE;
    const bool owns_internal = flags & Mem::OWNS_INTERNAL;
@@ -1018,7 +1018,7 @@ void *MemoryManager::ReadWrite_(void *h_ptr, MemoryType h_mt, MemoryClass mc,
                                 size_t bytes, unsigned &flags)
 {
    if (h_ptr) { CheckHostMemoryType_(h_mt, h_ptr, flags & Mem::ALIAS); }
-   if (bytes > 0) { MFEM_VERIFY(flags & Mem::REGISTERED,""); }
+   if (bytes > 0) { MFEM_VERIFY(flags & Mem::REGISTERED_,""); }
    MFEM_ASSERT(MemoryClassCheck_(mc, h_ptr, h_mt, bytes, flags),"");
    if (IsHostMemory(GetMemoryType(mc)) && mc < MemoryClass::DEVICE)
    {
@@ -1042,7 +1042,7 @@ const void *MemoryManager::Read_(void *h_ptr, MemoryType h_mt, MemoryClass mc,
                                  size_t bytes, unsigned &flags)
 {
    if (h_ptr) { CheckHostMemoryType_(h_mt, h_ptr, flags & Mem::ALIAS); }
-   if (bytes > 0) { MFEM_VERIFY(flags & Mem::REGISTERED,""); }
+   if (bytes > 0) { MFEM_VERIFY(flags & Mem::REGISTERED_,""); }
    MFEM_ASSERT(MemoryClassCheck_(mc, h_ptr, h_mt, bytes, flags),"");
    if (IsHostMemory(GetMemoryType(mc)) && mc < MemoryClass::DEVICE)
    {
@@ -1066,7 +1066,7 @@ void *MemoryManager::Write_(void *h_ptr, MemoryType h_mt, MemoryClass mc,
                             size_t bytes, unsigned &flags)
 {
    if (h_ptr) { CheckHostMemoryType_(h_mt, h_ptr, flags & Mem::ALIAS); }
-   if (bytes > 0) { MFEM_VERIFY(flags & Mem::REGISTERED,""); }
+   if (bytes > 0) { MFEM_VERIFY(flags & Mem::REGISTERED_,""); }
    MFEM_ASSERT(MemoryClassCheck_(mc, h_ptr, h_mt, bytes, flags),"");
    if (IsHostMemory(GetMemoryType(mc)) && mc < MemoryClass::DEVICE)
    {
@@ -1088,8 +1088,8 @@ void MemoryManager::SyncAlias_(const void *base_h_ptr, void *alias_h_ptr,
                                size_t alias_bytes, unsigned base_flags,
                                unsigned &alias_flags)
 {
-   // This is called only when (base_flags & Mem::REGISTERED) is true.
-   // Note that (alias_flags & REGISTERED) may not be true.
+   // This is called only when (base_flags & Mem::REGISTERED_) is true.
+   // Note that (alias_flags & REGISTERED_) may not be true.
    MFEM_ASSERT(alias_flags & Mem::ALIAS, "not an alias");
    if ((base_flags & Mem::VALID_HOST) && !(alias_flags & Mem::VALID_HOST))
    {
@@ -1097,10 +1097,10 @@ void MemoryManager::SyncAlias_(const void *base_h_ptr, void *alias_h_ptr,
    }
    if ((base_flags & Mem::VALID_DEVICE) && !(alias_flags & Mem::VALID_DEVICE))
    {
-      if (!(alias_flags & Mem::REGISTERED))
+      if (!(alias_flags & Mem::REGISTERED_))
       {
          mm.InsertAlias(base_h_ptr, alias_h_ptr, alias_bytes, base_flags & Mem::ALIAS);
-         alias_flags = (alias_flags | Mem::REGISTERED | Mem::OWNS_INTERNAL) &
+         alias_flags = (alias_flags | Mem::REGISTERED_ | Mem::OWNS_INTERNAL) &
                        ~(Mem::OWNS_HOST | Mem::OWNS_DEVICE);
       }
       mm.GetAliasDevicePtr(alias_h_ptr, alias_bytes, true);
@@ -1671,7 +1671,7 @@ void MemoryPrintFlags(unsigned flags)
 {
    typedef Memory<int> Mem;
    mfem::out
-         << "\n   registered    = " << bool(flags & Mem::REGISTERED)
+         << "\n   registered    = " << bool(flags & Mem::REGISTERED_)
          << "\n   owns host     = " << bool(flags & Mem::OWNS_HOST)
          << "\n   owns device   = " << bool(flags & Mem::OWNS_DEVICE)
          << "\n   owns internal = " << bool(flags & Mem::OWNS_INTERNAL)

--- a/general/mem_manager.hpp
+++ b/general/mem_manager.hpp
@@ -165,7 +165,15 @@ protected:
 
    enum FlagMask: unsigned
    {
+     // Workaround for use with headers that define REGISTERED as a macro,
+     // e.g. nb30.h (which is included by Windows.h):
+#ifndef REGISTERED
       REGISTERED    = 1 << 0, /**< The host pointer is registered with the
+                                   MemoryManager */
+#endif
+      // Use the following identifier if REGISTERED is defined as a macro,
+      // e.g. nb30.h (which is included by Windows.h):
+      REGISTERED_   = 1 << 0, /**< The host pointer is registered with the
                                    MemoryManager */
       OWNS_HOST     = 1 << 1, ///< The host pointer will be deleted by Delete()
       OWNS_DEVICE   = 1 << 2, /**< The device pointer will be deleted by
@@ -974,7 +982,7 @@ inline void Memory<T>::MakeAlias(const Memory &base, int offset, int size)
    capacity = size;
    h_mt = base.h_mt;
    h_ptr = base.h_ptr + offset;
-   if (!(base.flags & REGISTERED))
+   if (!(base.flags & REGISTERED_))
    {
       if (
 #if !defined(HYPRE_USING_GPU)
@@ -1010,7 +1018,7 @@ template <typename T>
 inline void Memory<T>::SetDeviceMemoryType(MemoryType d_mt)
 {
    if (!IsDeviceMemory(d_mt)) { return; }
-   if (!(flags & REGISTERED))
+   if (!(flags & REGISTERED_))
    {
       MemoryManager::Register_(h_ptr, nullptr, capacity*sizeof(T), h_mt,
                                flags & OWNS_HOST, flags & ALIAS, flags);
@@ -1021,7 +1029,7 @@ inline void Memory<T>::SetDeviceMemoryType(MemoryType d_mt)
 template <typename T>
 inline void Memory<T>::Delete()
 {
-   const bool registered = flags & REGISTERED;
+   const bool registered = flags & REGISTERED_;
    const bool mt_host = h_mt == MemoryType::HOST;
    const bool std_delete = !registered && mt_host;
 
@@ -1040,7 +1048,7 @@ inline void Memory<T>::Delete()
 template <typename T>
 inline void Memory<T>::DeleteDevice(bool copy_to_host)
 {
-   if (flags & REGISTERED)
+   if (flags & REGISTERED_)
    {
       if (copy_to_host) { Read(MemoryClass::HOST, capacity); }
       MemoryManager::DeleteDevice_((void*)h_ptr, flags);
@@ -1100,7 +1108,7 @@ template <typename T>
 inline T *Memory<T>::ReadWrite(MemoryClass mc, int size)
 {
    const size_t bytes = size * sizeof(T);
-   if (!(flags & REGISTERED))
+   if (!(flags & REGISTERED_))
    {
       if (mc == MemoryClass::HOST) { return h_ptr; }
       MemoryManager::Register_(h_ptr, nullptr, capacity*sizeof(T), h_mt,
@@ -1113,7 +1121,7 @@ template <typename T>
 inline const T *Memory<T>::Read(MemoryClass mc, int size) const
 {
    const size_t bytes = size * sizeof(T);
-   if (!(flags & REGISTERED))
+   if (!(flags & REGISTERED_))
    {
       if (mc == MemoryClass::HOST) { return h_ptr; }
       MemoryManager::Register_(h_ptr, nullptr, capacity*sizeof(T), h_mt,
@@ -1126,7 +1134,7 @@ template <typename T>
 inline T *Memory<T>::Write(MemoryClass mc, int size)
 {
    const size_t bytes = size * sizeof(T);
-   if (!(flags & REGISTERED))
+   if (!(flags & REGISTERED_))
    {
       if (mc == MemoryClass::HOST) { return h_ptr; }
       MemoryManager::Register_(h_ptr, nullptr, capacity*sizeof(T), h_mt,
@@ -1138,12 +1146,12 @@ inline T *Memory<T>::Write(MemoryClass mc, int size)
 template <typename T>
 inline void Memory<T>::Sync(const Memory &other) const
 {
-   if (!(flags & REGISTERED) && (other.flags & REGISTERED))
+   if (!(flags & REGISTERED_) && (other.flags & REGISTERED_))
    {
       MFEM_ASSERT(h_ptr == other.h_ptr &&
                   (flags & ALIAS) == (other.flags & ALIAS),
                   "invalid input");
-      flags = (flags | REGISTERED) & ~(OWNS_DEVICE | OWNS_INTERNAL);
+      flags = (flags | REGISTERED_) & ~(OWNS_DEVICE | OWNS_INTERNAL);
    }
    flags = (flags & ~(VALID_HOST | VALID_DEVICE)) |
            (other.flags & (VALID_HOST | VALID_DEVICE));
@@ -1153,9 +1161,9 @@ template <typename T>
 inline void Memory<T>::SyncAlias(const Memory &base, int alias_size) const
 {
    // Assuming that if *this is registered then base is also registered.
-   MFEM_ASSERT(!(flags & REGISTERED) || (base.flags & REGISTERED),
+   MFEM_ASSERT(!(flags & REGISTERED_) || (base.flags & REGISTERED_),
                "invalid base state");
-   if (!(base.flags & REGISTERED)) { return; }
+   if (!(base.flags & REGISTERED_)) { return; }
    MemoryManager::SyncAlias_(base.h_ptr, h_ptr, alias_size*sizeof(T),
                              base.flags, flags);
 }
@@ -1170,7 +1178,7 @@ inline MemoryType Memory<T>::GetMemoryType() const
 template <typename T>
 inline MemoryType Memory<T>::GetDeviceMemoryType() const
 {
-   if (!(flags & REGISTERED)) { return MemoryType::DEFAULT; }
+   if (!(flags & REGISTERED_)) { return MemoryType::DEFAULT; }
    return MemoryManager::GetDeviceMemoryType_(h_ptr, flags & ALIAS);
 }
 
@@ -1190,7 +1198,7 @@ template <typename T>
 inline void Memory<T>::CopyFrom(const Memory &src, int size)
 {
    MFEM_VERIFY(src.capacity>=size && capacity>=size, "Incorrect size");
-   if (!(flags & REGISTERED) && !(src.flags & REGISTERED))
+   if (!(flags & REGISTERED_) && !(src.flags & REGISTERED_))
    {
       if (h_ptr != src.h_ptr && size != 0)
       {
@@ -1210,7 +1218,7 @@ template <typename T>
 inline void Memory<T>::CopyFromHost(const T *src, int size)
 {
    MFEM_VERIFY(capacity>=size, "Incorrect size");
-   if (!(flags & REGISTERED))
+   if (!(flags & REGISTERED_))
    {
       if (h_ptr != src && size != 0)
       {
@@ -1237,7 +1245,7 @@ template <typename T>
 inline void Memory<T>::CopyToHost(T *dest, int size) const
 {
    MFEM_VERIFY(capacity>=size, "Incorrect size");
-   if (!(flags & REGISTERED))
+   if (!(flags & REGISTERED_))
    {
       if (h_ptr != dest && size != 0)
       {

--- a/general/mem_manager.hpp
+++ b/general/mem_manager.hpp
@@ -173,7 +173,7 @@ protected:
 #endif
       // Use the following identifier if REGISTERED is defined as a macro,
       // e.g. nb30.h (which is included by Windows.h):
-      REGISTERED_   = 1 << 0, /**< The host pointer is registered with the
+      Registered    = 1 << 0, /**< The host pointer is registered with the
                                    MemoryManager */
       OWNS_HOST     = 1 << 1, ///< The host pointer will be deleted by Delete()
       OWNS_DEVICE   = 1 << 2, /**< The device pointer will be deleted by
@@ -982,7 +982,7 @@ inline void Memory<T>::MakeAlias(const Memory &base, int offset, int size)
    capacity = size;
    h_mt = base.h_mt;
    h_ptr = base.h_ptr + offset;
-   if (!(base.flags & REGISTERED_))
+   if (!(base.flags & Registered))
    {
       if (
 #if !defined(HYPRE_USING_GPU)
@@ -1018,7 +1018,7 @@ template <typename T>
 inline void Memory<T>::SetDeviceMemoryType(MemoryType d_mt)
 {
    if (!IsDeviceMemory(d_mt)) { return; }
-   if (!(flags & REGISTERED_))
+   if (!(flags & Registered))
    {
       MemoryManager::Register_(h_ptr, nullptr, capacity*sizeof(T), h_mt,
                                flags & OWNS_HOST, flags & ALIAS, flags);
@@ -1029,7 +1029,7 @@ inline void Memory<T>::SetDeviceMemoryType(MemoryType d_mt)
 template <typename T>
 inline void Memory<T>::Delete()
 {
-   const bool registered = flags & REGISTERED_;
+   const bool registered = flags & Registered;
    const bool mt_host = h_mt == MemoryType::HOST;
    const bool std_delete = !registered && mt_host;
 
@@ -1048,7 +1048,7 @@ inline void Memory<T>::Delete()
 template <typename T>
 inline void Memory<T>::DeleteDevice(bool copy_to_host)
 {
-   if (flags & REGISTERED_)
+   if (flags & Registered)
    {
       if (copy_to_host) { Read(MemoryClass::HOST, capacity); }
       MemoryManager::DeleteDevice_((void*)h_ptr, flags);
@@ -1108,7 +1108,7 @@ template <typename T>
 inline T *Memory<T>::ReadWrite(MemoryClass mc, int size)
 {
    const size_t bytes = size * sizeof(T);
-   if (!(flags & REGISTERED_))
+   if (!(flags & Registered))
    {
       if (mc == MemoryClass::HOST) { return h_ptr; }
       MemoryManager::Register_(h_ptr, nullptr, capacity*sizeof(T), h_mt,
@@ -1121,7 +1121,7 @@ template <typename T>
 inline const T *Memory<T>::Read(MemoryClass mc, int size) const
 {
    const size_t bytes = size * sizeof(T);
-   if (!(flags & REGISTERED_))
+   if (!(flags & Registered))
    {
       if (mc == MemoryClass::HOST) { return h_ptr; }
       MemoryManager::Register_(h_ptr, nullptr, capacity*sizeof(T), h_mt,
@@ -1134,7 +1134,7 @@ template <typename T>
 inline T *Memory<T>::Write(MemoryClass mc, int size)
 {
    const size_t bytes = size * sizeof(T);
-   if (!(flags & REGISTERED_))
+   if (!(flags & Registered))
    {
       if (mc == MemoryClass::HOST) { return h_ptr; }
       MemoryManager::Register_(h_ptr, nullptr, capacity*sizeof(T), h_mt,
@@ -1146,12 +1146,12 @@ inline T *Memory<T>::Write(MemoryClass mc, int size)
 template <typename T>
 inline void Memory<T>::Sync(const Memory &other) const
 {
-   if (!(flags & REGISTERED_) && (other.flags & REGISTERED_))
+   if (!(flags & Registered) && (other.flags & Registered))
    {
       MFEM_ASSERT(h_ptr == other.h_ptr &&
                   (flags & ALIAS) == (other.flags & ALIAS),
                   "invalid input");
-      flags = (flags | REGISTERED_) & ~(OWNS_DEVICE | OWNS_INTERNAL);
+      flags = (flags | Registered) & ~(OWNS_DEVICE | OWNS_INTERNAL);
    }
    flags = (flags & ~(VALID_HOST | VALID_DEVICE)) |
            (other.flags & (VALID_HOST | VALID_DEVICE));
@@ -1161,9 +1161,9 @@ template <typename T>
 inline void Memory<T>::SyncAlias(const Memory &base, int alias_size) const
 {
    // Assuming that if *this is registered then base is also registered.
-   MFEM_ASSERT(!(flags & REGISTERED_) || (base.flags & REGISTERED_),
+   MFEM_ASSERT(!(flags & Registered) || (base.flags & Registered),
                "invalid base state");
-   if (!(base.flags & REGISTERED_)) { return; }
+   if (!(base.flags & Registered)) { return; }
    MemoryManager::SyncAlias_(base.h_ptr, h_ptr, alias_size*sizeof(T),
                              base.flags, flags);
 }
@@ -1178,7 +1178,7 @@ inline MemoryType Memory<T>::GetMemoryType() const
 template <typename T>
 inline MemoryType Memory<T>::GetDeviceMemoryType() const
 {
-   if (!(flags & REGISTERED_)) { return MemoryType::DEFAULT; }
+   if (!(flags & Registered)) { return MemoryType::DEFAULT; }
    return MemoryManager::GetDeviceMemoryType_(h_ptr, flags & ALIAS);
 }
 
@@ -1198,7 +1198,7 @@ template <typename T>
 inline void Memory<T>::CopyFrom(const Memory &src, int size)
 {
    MFEM_VERIFY(src.capacity>=size && capacity>=size, "Incorrect size");
-   if (!(flags & REGISTERED_) && !(src.flags & REGISTERED_))
+   if (!(flags & Registered) && !(src.flags & Registered))
    {
       if (h_ptr != src.h_ptr && size != 0)
       {
@@ -1218,7 +1218,7 @@ template <typename T>
 inline void Memory<T>::CopyFromHost(const T *src, int size)
 {
    MFEM_VERIFY(capacity>=size, "Incorrect size");
-   if (!(flags & REGISTERED_))
+   if (!(flags & Registered))
    {
       if (h_ptr != src && size != 0)
       {
@@ -1245,7 +1245,7 @@ template <typename T>
 inline void Memory<T>::CopyToHost(T *dest, int size) const
 {
    MFEM_VERIFY(capacity>=size, "Incorrect size");
-   if (!(flags & REGISTERED_))
+   if (!(flags & Registered))
    {
       if (h_ptr != dest && size != 0)
       {

--- a/general/mem_manager.hpp
+++ b/general/mem_manager.hpp
@@ -165,8 +165,8 @@ protected:
 
    enum FlagMask: unsigned
    {
-     // Workaround for use with headers that define REGISTERED as a macro,
-     // e.g. nb30.h (which is included by Windows.h):
+      // Workaround for use with headers that define REGISTERED as a macro,
+      // e.g. nb30.h (which is included by Windows.h):
 #ifndef REGISTERED
       REGISTERED    = 1 << 0, /**< The host pointer is registered with the
                                    MemoryManager */

--- a/miniapps/tools/convert-dc.cpp
+++ b/miniapps/tools/convert-dc.cpp
@@ -190,7 +190,7 @@ int main(int argc, char *argv[])
    src_dc->SetPadDigitsRank(src_pad_digits_rank);
    src_dc->Load(src_cycle);
 
-   if (src_dc->Error() != DataCollection::NO_ERROR_)
+   if (src_dc->Error() != DataCollection::No_Error)
    {
       mfem::out << "Error loading data collection: "
                 << src_coll_name
@@ -227,7 +227,7 @@ int main(int argc, char *argv[])
 
    out_dc->Save();
 
-   if (out_dc->Error() != DataCollection::NO_ERROR_)
+   if (out_dc->Error() != DataCollection::No_Error)
    {
       mfem::out << "Error saving data collection: "
                 << out_coll_name

--- a/miniapps/tools/convert-dc.cpp
+++ b/miniapps/tools/convert-dc.cpp
@@ -190,7 +190,7 @@ int main(int argc, char *argv[])
    src_dc->SetPadDigitsRank(src_pad_digits_rank);
    src_dc->Load(src_cycle);
 
-   if (src_dc->Error() != DataCollection::NO_ERROR)
+   if (src_dc->Error() != DataCollection::NO_ERROR_)
    {
       mfem::out << "Error loading data collection: "
                 << src_coll_name
@@ -227,7 +227,7 @@ int main(int argc, char *argv[])
 
    out_dc->Save();
 
-   if (out_dc->Error() != DataCollection::NO_ERROR)
+   if (out_dc->Error() != DataCollection::NO_ERROR_)
    {
       mfem::out << "Error saving data collection: "
                 << out_coll_name

--- a/miniapps/tools/get-values.cpp
+++ b/miniapps/tools/get-values.cpp
@@ -117,7 +117,7 @@ int main(int argc, char *argv[])
    dc.SetPadDigitsRank(pad_digits_rank);
    dc.Load(cycle);
 
-   if (dc.Error() != DataCollection::NO_ERROR_)
+   if (dc.Error() != DataCollection::No_Error)
    {
       mfem::out << "Error loading VisIt data collection: " << coll_name << endl;
       return 1;

--- a/miniapps/tools/get-values.cpp
+++ b/miniapps/tools/get-values.cpp
@@ -117,7 +117,7 @@ int main(int argc, char *argv[])
    dc.SetPadDigitsRank(pad_digits_rank);
    dc.Load(cycle);
 
-   if (dc.Error() != DataCollection::NO_ERROR)
+   if (dc.Error() != DataCollection::NO_ERROR_)
    {
       mfem::out << "Error loading VisIt data collection: " << coll_name << endl;
       return 1;

--- a/miniapps/tools/load-dc.cpp
+++ b/miniapps/tools/load-dc.cpp
@@ -73,7 +73,7 @@ int main(int argc, char *argv[])
    dc.SetPadDigitsRank(pad_digits_rank);
    dc.Load(cycle);
 
-   if (dc.Error() != DataCollection::NO_ERROR_)
+   if (dc.Error() != DataCollection::No_Error)
    {
       mfem::out << "Error loading VisIt data collection: " << coll_name << endl;
       return 1;

--- a/miniapps/tools/load-dc.cpp
+++ b/miniapps/tools/load-dc.cpp
@@ -73,7 +73,7 @@ int main(int argc, char *argv[])
    dc.SetPadDigitsRank(pad_digits_rank);
    dc.Load(cycle);
 
-   if (dc.Error() != DataCollection::NO_ERROR)
+   if (dc.Error() != DataCollection::NO_ERROR_)
    {
       mfem::out << "Error loading VisIt data collection: " << coll_name << endl;
       return 1;

--- a/tests/unit/general/test_mem.cpp
+++ b/tests/unit/general/test_mem.cpp
@@ -44,8 +44,8 @@ TEST_CASE("MemoryManager/Scopes",
       REQUIRE(h_x == x.GetData());
       REQUIRE(mem->capacity == x.Size());
       REQUIRE(mem->h_mt == Device::GetHostMemoryType());
-      constexpr unsigned REGISTERED_ = 1 << 0;
-      const bool registered = mem->flags & REGISTERED_;
+      constexpr unsigned Registered = 1 << 0;
+      const bool registered = mem->flags & Registered;
       const bool registered_is_known = registered == mm.IsKnown(h_x);
       REQUIRE(registered_is_known);
    }

--- a/tests/unit/general/test_mem.cpp
+++ b/tests/unit/general/test_mem.cpp
@@ -44,8 +44,8 @@ TEST_CASE("MemoryManager/Scopes",
       REQUIRE(h_x == x.GetData());
       REQUIRE(mem->capacity == x.Size());
       REQUIRE(mem->h_mt == Device::GetHostMemoryType());
-      constexpr unsigned REGISTERED = 1 << 0;
-      const bool registered = mem->flags & REGISTERED;
+      constexpr unsigned REGISTERED_ = 1 << 0;
+      const bool registered = mem->flags & REGISTERED_;
       const bool registered_is_known = registered == mm.IsKnown(h_x);
       REQUIRE(registered_is_known);
    }

